intro/10_scope/basic_game.rb

```diff
@@ -1,9 +1,35 @@
 class Player
   # Your code goes here
+    def initialize(name)
+        @name = name
+    end
 end
 
 class Game
   # Your code goes here
+    @@MAX_PLAYER_COUNT = 0  
+
+    attr_accessor :players
+
+    def initialize(name)
+        @name = name
+        @players = Array.new
+    end
+
+    def Game.MAX_PLAYER_COUNT
+        @@MAX_PLAYER_COUNT
+    end
+
+    def add_player(player)
```

Comment:
It seems like you might have slightly misunderstood the purpose of `MAX_PLAYER_COUNT` (and I might not have explained it well enough). When adding a player to the game, you should first check to see if there are already the max number of players. If so, do not add the next player (perhaps log an error or tell the user why it didn't work).